eth/db/header.py

```diff
@@ -204,62 +205,78 @@ def persist_header_chain(self,
         with self.db.atomic_batch() as db:
             return self._persist_header_chain(db, headers)
 
+    @classmethod
+    def _set_hash_scores_to_db(
+            cls,
+            db: BaseDB,
+            header: BlockHeader,
+            score: int
+    ) -> int:
+        db.set(
+            header.hash,
+            rlp.encode(header),
+        )
```

Comment:
This feels odd.  Why does this function also persist the header itself?

Reply:
To give a context, the code is coming from this line: https://github.com/ethereum/py-evm/pull/1701/files#diff-ad4737f864351f38187efa966585825eL241

Reply:
With the current name of the function `_set_hash_scores_to_db` this code doesn't belong because persisting the header itself isn't the same as persisting the header's total difficulty/score.


I'd be fine with either 1) renaming this function to something like `_persist_header_to_db` or 2) just inlining this functionality in the two places it is needed within the `persist_header_chain` function.

Preference is currently for option 2

Reply:
Thanks for your input. I went for option 2.